textBasedGameRedo.js

```diff
@@ -0,0 +1,232 @@
+            takeMeds()
+        } else {
+            vomit()
+        }
+    } else {
+        finishGame();
+    }
+}
+
+const vomit = () => {
+    console.log("You feel as if you're going to throw up.")
+    console.log("Before you can react, you throw up on your laptop!")
+    console.log("All your data is gone.")
+    endGame()
+}
+
+const takeMeds = () => {
+    nearDeathCount++
+    console.log("You run to your bathroom cabinet and grab medication.")
+    let choices = ["Take recommended dose", "Take what you feel is right" ]
+    let userInput = readline.keyInSelect(choices, "Do you:" )
+
+    switch (userInput){
+        case 0: 
+            finishGame()
+        case 1:
+            hallucination()
+
+    }
+    
+}
+
+const hallucination = () => {
+    console.log("You see a silhouette appear in the dark across from you")
+    let choices = ["Walk towards it", "Run away", "Throw a brick"]
+    let userInput = readline.keyInSelect(choices, "What action do you take?")
+
+    switch (userInput){
+        case 0:
+            walkTowardsIt()
+        case 1: 
+            runAway()
+        case 2:
+            throwBrick()
+    }
+}
+
+const walkTowardsIt = () => {
+    console.log("As you walk towards it the shadow disappears")
+    console.log("It hits you and you collapse")
+    console.log("As your eyes close for the last time, you see the shadow standing over you")
+    endGame()
+}
+
+const runAway = () => {
+    console.log("You leave your apartment and try to find help from your next door neighbor.")
+    console.log("Your neighbors call the police and you are sent to a mental asylum.")
+    endGame()
+}
+
+const throwBrick = () => {
+    nearDeathCount++
+    console.log("The shadow screeches")
+    console.log("The shadow grabs the brick and throws it back")
+    
+    let choices = ["Run away", "Turn on the lights", "Splash water on face"]
+    let userInput = readline.keyInSelect(choices,"What action do you take?" )
+
+    switch(userInput){
+        case 0: 
+            runAway()
+        case 1:
+            turnOnLight()
+        case 2:
+            splashWater()
+    }
+}
+
+const turnOnLight = () => {
+    nearDeathCount++
+    console.log("The silhouette does not disappear")
+    console.log("It hits you and you collapse")
+    console.log("As your eyes close for the last time, you see the silhouette standing over you")
+    endGame()
+    process.exit()
+}
+
+const splashWater = () => {
+    nearDeathCount++
+    console.log("You return to your normal senses.")
+    console.log("You sit back down at your desk to continue your game.")
+    finishGame()
+    process.exit()
+}
+
+const finishGame = () => {
+    console.log("Congrats! You made it to the end alive")
+    for(let i = 0; i <= nearDeathCount; i++ ){
+        console.log("You have survived! " + i)
+    }
+    process.exit()
+}
+
+const endGame = () => {
+    console.log("You weren't able to stay alive")
```

Comment:
you can replace endGame and finishGame with one function and pass an argument to select which console log the player to see.